.github/workflows/dco.yml

```diff
@@ -0,0 +1,30 @@
+name: "DCO Assistant"
+on:
+  issue_comment:
+    types: [created]
+  pull_request_target:
+    types: [opened, synchronize]
+
+jobs:
+  DCO:
+    runs-on: ubuntu-latest
+    steps:
+      - name: "DCO Assistant"
+        if: (github.event.comment.body == 'recheck' || github.event.comment.body == 'I have read the DCO document and I hereby sign the DCO.') || github.event_name == 'pull_request_target'
+        uses: cla-assistant/github-action@6cfe025afc31eb0bdd08e23d3f2e9ed141ab2c09
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PERSONAL_ACCESS_TOKEN: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+        with:
+          path-to-signatures: "dco-signatures.json"
+          path-to-document: "https://github.com/carbon-design-system/carbon-dco/blob/main/dco.md"
+          branch: "main"
+          allowlist: dependabot,kodiakhq
```

Comment:
Will Vercel also trigger this when it comments the deploy previews?

What about PR's by another action like https://github.com/carbon-design-system/carbon-website/pull/2177

Reply:
`kodiakhq` bot makes the commits for the image compression PRs, so I think we're good there. E.g. https://github.com/carbon-design-system/carbon-website/pull/2177/commits/2b48d3b5a44e16d169c49ce0135ce4757594398e

I'm unsure if this check runs on every comment like Vercel deploy previews. If the action is slowing down our review, we can always disable it and look for alternative solutions.

Reply:
Yeah, just something I was thinking of, definitely not a blocker. Thanks for adding this in!